### PR TITLE
chore: use a pair of bigints to represent ratio

### PR DESCRIPTION
**Pull request by rabi-siddique**
## Description

- Replaces the floating-point representation of ratios with a pair of BigInt values.
- Adds a test to validate gas ratio in `offerArgs`.
- Enables and fixes the test for opening a position on Compound. 

### Security Considerations

N/A
### Scaling Considerations

N/A
### Documentation Considerations

N/A
### Testing Considerations

The vstorage snapshot tests should be enough to verify this change, which are passing. 

### Upgrade Considerations

N/A

### CODE REVIEW

File: packages/portfolio-contract/src/portfolio.flows.ts [hunk 1]
```diff
@@ -205,7 +205,10 @@ export const makeAxelarMemo = (
 
   memo.fee = {
     // This amount specifies the outbound gas for sending GMP message
-    amount: String(gasRatio * Number(gasAmounts[keyword].value)),
+    amount: String(
+      (Number(gasRatio[0]) / Number(gasRatio[1])) *
```
Comment (amessbee) [inline packages/portfolio-contract/src/portfolio.flows.ts:209]: Make sure that `Number(gasRatio[1])` is not zero.

File: packages/portfolio-contract/src/type-guards.ts [hunk 2]
```diff
@@ -96,32 +96,32 @@ export const makeProposalShapes = (
   };
 };
 
+type Ratio = [numerator: bigint, denominator: bigint];
```
Comment (amessbee) [inline packages/portfolio-contract/src/type-guards.ts]: For code readability maybe disambiguate it from Zoe data type Ratio.

File: packages/portfolio-contract/src/type-guards.ts [hunk 3]
```diff
@@ -135,10 +135,8 @@ const RatioShape = M.opt(
   M.splitRecord(
     {},
     {
-      // TODO: use pair of big ints
-      // maybe use RatioShape from ertp package?
-      gmpRatio: M.and(M.gt(0), M.lt(1)),
-      acctRatio: M.and(M.gt(0), M.lt(1)),
+      gmpRatio: M.splitArray([M.nat(), M.nat()]),
```
Comment (amessbee) [inline packages/portfolio-contract/src/type-guards.ts]: Also check that numerator is less than or equal to denominator other than ratio would be more than 1.

File: packages/portfolio-contract/test/portfolio.contract.test.ts [hunk 4]
```diff
@@ -225,7 +225,7 @@ test('makeAxelarMemo constructs correct memo JSON', t => {
   const type = 2; // contract call with tokens
   const destinationEVMChain = 'Avalanche';
   const destinationAddress = '0x58E0bd49520364A115CeE4B03DffC1C08A2D1D09';
-  const gasRatio = 0.5;
+  const gasRatio: [bigint, bigint] = [1n, 2n];
```
Comment (amessbee) [inline packages/portfolio-contract/test/portfolio.contract.test.ts]: maybe use the Ratio type here as well.

File: packages/portfolio-contract/src/type-guards.ts (hunk 3 above)
Comment (amessbee) [reply packages/portfolio-contract/src/type-guards.ts]: Also, add a test for it.

File: packages/portfolio-contract/src/portfolio.flows.ts (hunk 1 above)
Comment (amessbee) [reply packages/portfolio-contract/src/portfolio.flows.ts:209]: Also, add a test for it - that the contract or zoe complains when `Number(gasRatio[1])` is  zero or less than `Number(gasRatio[0])`.

**Review by amessbee — APPROVED** [review]
Fix the minor comments - otherwise looks good.
